core/templates/services/search.service.ts

```diff
@@ -16,215 +16,234 @@
-    // Appends a suffix to the query describing allowed category and language
-    // codes to filter on.
-    var _getSuffixForQuery =
-      function(selectedCategories, selectedLanguageCodes) {
-        var querySuffix = '';
-
-        var _categories = '';
-        for (var key in selectedCategories) {
-          if (selectedCategories[key]) {
-            if (_categories) {
-              _categories += '" OR "';
-            }
-            _categories += key;
-          }
-        }
-        if (_categories) {
-          querySuffix += '&category=("' + _categories + '")';
-        }
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable, EventEmitter } from '@angular/core';
+
+import { ServicesConstants } from 'services/services.constants';
+import { ExplorationSummaryBackendDict } from 'domain/summary/exploration-summary-backend-api.service';
+import { SearchBackendApiService, SelectionDetails, SelectionList } from './search-backend-api.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchService {
+  private lastQuery: string = null;
+  private lastSelectedCategories: SelectionList = {};
+  private lastSelectedLanguageCodes: SelectionList = {};
+  private searchCursor: string = null;
+  private isCurrentlyFetchingResults = false;
+  private searchBarLoadedEventEmitter = new EventEmitter<string>();
+  private initialSearchResultsLoadedEventEmitter =
+    new EventEmitter<ExplorationSummaryBackendDict>();
+  public numSearchesInProgress = 0;
+
+  constructor(
+    private searchBackendApiService: SearchBackendApiService) {
+  }
 
-        var _languageCodes = '';
-        for (var key in selectedLanguageCodes) {
-          if (selectedLanguageCodes[key]) {
-            if (_languageCodes) {
-              _languageCodes += '" OR "';
-            }
-            _languageCodes += key;
-          }
+  private getSuffixForQuery(
+      selectedCategories:SelectionList,
+      selectedLanguageCodes:SelectionList) {
+    let querySuffix = '';
+
+    let _categories = '';
+    for (let key in selectedCategories) {
+      if (selectedCategories[key]) {
+        if (_categories) {
+          _categories += '" OR "';
         }
+        _categories += key;
+      }
+    }
+    if (_categories) {
+      querySuffix += '&category=("' + _categories + '")';
+    }
+
+    let _languageCodes = '';
+    for (let key in selectedLanguageCodes) {
+      if (selectedLanguageCodes[key]) {
         if (_languageCodes) {
-          querySuffix += '&language_code=("' + _languageCodes + '")';
+          _languageCodes += '" OR "';
         }
+        _languageCodes += key;
+      }
+    }
+    if (_languageCodes) {
+      querySuffix += '&language_code=("' + _languageCodes + '")';
+    }
 
-        return querySuffix;
-      };
+    return querySuffix;
+  }
 
-    var hasReachedEndOfPage = function() {
-      return _searchCursor === null;
-    };
+  hasReachedEndOfPage():boolean {
```

Comment:
Add a space after `:`. Also do this for other function here.
```suggestion
  hasReachedEndOfPage(): boolean {
```

Reply:
done